File: backend/src/MalihaPolytex.Domain/Entities/JobApplication.cs
namespace MalihaPolytex.Domain.Entities;

public class JobApplication
{
    public int JobApplicationId { get; set; }

    public int JobId { get; set; }

    public string JobTitle { get; set; } = string.Empty;

    public string FullName { get; set; } = string.Empty;

    public string Email { get; set; } = string.Empty;

    public string Phone { get; set; } = string.Empty;

    public string? Address { get; set; }

    public string? HighestEducation { get; set; }

    public decimal? YearsOfExperience { get; set; }

    public string? CurrentCompany { get; set; }

    public string? CurrentDesignation { get; set; }

    public string? ExpectedSalary { get; set; }

    public string? CoverLetter { get; set; }

    public string ResumeFile { get; set; } = string.Empty;

    public byte StatusId { get; set; }

    public string? Remarks { get; set; }

    public DateTime AppliedDate { get; set; }

    public DateTime CreatedDate { get; set; }

    public string? CreatedBy { get; set; }

    public DateTime? ModifiedDate { get; set; }

    public string? ModifiedBy { get; set; }
    public string StatusName { get; set; } = string.Empty;
}

File: admin/src/types/jobApplication.ts
export interface JobApplication {
  jobApplicationId: number;
  jobId: number;
  jobTitle: string;
  fullName: string;
  email: string;
  phone: string;
  address?: string;
  highestEducation?: string;
  yearsOfExperience?: number;
  currentCompany?: string;
  currentDesignation?: string;
  expectedSalary?: string;
  coverLetter?: string;
  resumeFile: string;
  statusId: number;
  statusName: string;
  remarks?: string;
  appliedDate: string;
  createdDate: string;
  createdBy?: string;
  modifiedDate?: string;
  modifiedBy?: string;
}

export interface JobApplicationStatusUpdateRequest {
  statusId: number;
  remarks?: string;
}

export interface CareerApplicationStatus {
  statusId: number;
  statusName: string;
}

export const careerApplicationStatuses: CareerApplicationStatus[] = [
  {
    statusId: 1,
    statusName: "Pending",
  },
  {
    statusId: 2,
    statusName: "Under Review",
  },
  {
    statusId: 3,
    statusName: "Shortlisted",
  },
  {
    statusId: 4,
    statusName: "Interview Scheduled",
  },
  {
    statusId: 5,
    statusName: "Hired",
  },
  {
    statusId: 6,
    statusName: "Rejected",
  },
  {
    statusId: 7,
    statusName: "Withdrawn",
  },
];

export interface JobApplicationSearchRequest {
  page: number;
  pageSize: number;
  search?: string;
  jobId?: number;
  statusId?: number;
  maxExperience?: number;
}

export interface PagedResult<T> {
  items: T[];
  page: number;
  pageSize: number;
  totalCount: number;
  totalPages: number;
}